### ISSUE TITLE
Examples

### ISSUE THREAD

**Issue by gmosx** [google-apis-rs/generator#24]
First of all, thank you, this is an extremely useful project!

I am wondering if there are any examples to demonstrate how to use the generated libraries. E.g. a small cli app that lists events from a Google calendar or something similar. 🙏

**Comment by gmosx** [comment]
I see there are some CLI apps in the `generated` repository, but it's not obvious how to setup authentication/authorization.

**Comment by mwilliammyers** [comment]
Yeah authentication is a tricky part of the setup. I can see if I can add an example, but in the meantime, we use  version `^3`of [yup-oauth2](https://github.com/dermesser/yup-oauth2). Most likely you want to use a [service account file](https://cloud.google.com/iam/docs/creating-managing-service-account-keys) to authenticate.

**Comment by mwilliammyers** [comment]
All that being said, I just noticed you wanted a Google calendar app, which isn't a Google Cloud API so I am sure the auth API will be different. My bad! 🙃 You will probably want to do an oauth2 flow for authenticating https://developers.google.com/identity/protocols/oauth2

**Comment by gmosx** [comment]
Hey William, thank you for the information. Will check it. Google Calendar was just an example, even though I think it is a Google API. In any case, will check yup-oauth2 as you suggested.

**Comment by braincow** [comment]
This example no longer compiles:
```
error[E0308]: mismatched types
   --> src/main.rs:11:9
    |
11  |         yup_oauth2::ServiceAccountAccess::new(credentials).build(),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `yup_oauth2::authenticator::Authenticator`, found opaque type
    | 
   ::: /home/bcow/.cargo/registry/src/github.com-1ecc6299db9ec823/yup-oauth2-3.1.1/src/service_account.rs:234:27
    |
234 |     pub fn build(self) -> impl GetToken {
    |                           ------------- the found opaque type
    |
    = note:   expected struct `yup_oauth2::authenticator::Authenticator<_>`
            found opaque type `impl yup_oauth2::types::GetToken`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `gcp-test`.
```

**Comment by kotx** [comment]
How would an API that only needs an API key be used?

**Comment by DazWilkin** [comment]
It would be very useful to have a reference implementation of one of the generated (e.g. Cloud) APIs to provide guidance to those of us noobs who struggle to use this repo.

I'm also affected by the auth example in this issue no longer compiling.

I tried to understand how drmesser's example [here](https://github.com/dermesser/yup-oauth2/blob/5d0e431772bdd7e3d42b8dfc56acbedc8dce2958/examples/service_account/src/main.rs#L186)

It appears that `yup_oauth2` no longer provides `access_token` but `access` and ... IIUC... this borks `google_api_auth`'s `GetAccessToken` trait.
